### PR TITLE
(PA-514) correct key to remembered props on i386 win

### PR DESCRIPTION
**Pull request by MosesMendoza**
Prior to this commit, beaker always assumed a specific registry key for the
remembered properties when validating them for an MSI on install. However, on
32 bit installs, the registry key we right to is different. This commit updates
the validation to search for the correct, different, key based on host
architecture.

Signed-off-by: Moses Mendoza <moses@puppet.com>

### CODE REVIEW

**Comment by MosesMendoza** [comment]
cc @Iristyle for review

**Comment by MosesMendoza** [comment]
Currently testing basic install on Windows 10 enterprise 32-bit

**Comment by MosesMendoza** [comment]
Need to update spec tests

File: lib/beaker/dsl/install_utils/windows_utils.rb [hunk 1]
```diff
@@ -143,7 +143,9 @@ def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
 
             # (PA-514) value for PUPPET_AGENT_STARTUP_MODE should be present in
             # registry and honored after install/upgrade.
-            reg_query_command = %Q(reg query "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" /v "RememberedPuppetAgentStartupMode" | findstr #{msi_opts['PUPPET_AGENT_STARTUP_MODE']})
+            reg_key = host.is_x86_64? ? "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" :
```
Comment (Iristyle) [inline lib/beaker/dsl/install_utils/windows_utils.rb]: This looks good conceptually... will have to run some tests locally to make sure we're gravy.

**Comment by Iristyle** [comment]
There was also one more comment at https://github.com/puppetlabs/beaker/pull/1239#discussion_r79680470

**Comment by MosesMendoza** [comment]
@Iristyle updated w/feedback re: `sc qc`. I think this is ready for further review / testing?

File: lib/beaker/dsl/install_utils/windows_utils.rb [hunk 2]
```diff
@@ -139,18 +139,34 @@ def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
             # if puppet service exists, then pe-puppet is not queried
             # if puppet service does not exist, pe-puppet is queried and that exit code is used
             # therefore, this command will always exit 0 if either service is installed
-            on host, Command.new("sc query puppet || sc query pe-puppet", [], { :cmdexe => true })
+            #
+            # We also take advantage of this output to verify the startup
+            # settings are honored as supplied to the MSI
+            on host, Command.new("sc qc puppet || sc qc pe-puppet", [], { :cmdexe => true }) do |result|
```
Comment (Iristyle) [inline lib/beaker/dsl/install_utils/windows_utils.rb]: 👍 Looking good

**Comment by MosesMendoza** [comment]
Derp. Forgot to update beaker specs for that latest change

**Comment by MosesMendoza** [comment]
Relevant output from this installing MSI on a Windows 10 Enterprise 32-bit host: https://gist.github.com/MosesMendoza/e82bdf4b59318f891ba5b01f09744ef2

**Comment by MosesMendoza** [comment]
Relevant output from this installing MSI (and later upgrading) with beaker on Windows 2008 64 bit: https://gist.github.com/MosesMendoza/30f79facfbd2429d6d9435eb2768c0b4

**Comment by Iristyle** [comment]
Previous output for puppet with `sc query`:

```
SERVICE_NAME: puppet
        TYPE               : 10  WIN32_OWN_PROCESS
        STATE              : 1  STOPPED
        WIN32_EXIT_CODE    : 1077  (0x435)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
```

New output with `sc qc puppet`:

```
    SERVICE_NAME: puppet
            TYPE               : 10  WIN32_OWN_PROCESS
            START_TYPE         : 3   DEMAND_START
            ERROR_CONTROL      : 1   NORMAL
            BINARY_PATH_NAME   : "C:\Program Files\Puppet Labs\Puppet\sys\ruby\bin\ruby.exe" -rubygems "C:\Program Files\Puppet Labs\Puppet\service\daemon.rb"
            LOAD_ORDER_GROUP   :
            TAG                : 0
            DISPLAY_NAME       : Puppet Agent
            DEPENDENCIES       :
            SERVICE_START_NAME : LocalSystem
```

Bonus!

File: lib/beaker/dsl/install_utils/windows_utils.rb [hunk 3]
```diff
@@ -139,18 +139,34 @@ def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
...
+                raise "puppet service startup mode did not match supplied MSI option '#{startup_mode}'"
+              end
+            end
 
             # (PA-514) value for PUPPET_AGENT_STARTUP_MODE should be present in
             # registry and honored after install/upgrade.
-            reg_query_command = %Q(reg query "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" /v "RememberedPuppetAgentStartupMode" | findstr #{msi_opts['PUPPET_AGENT_STARTUP_MODE']})
+            reg_key = host.is_x86_64? ? "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" :
+                                        "HKLM\\SOFTWARE\\Puppet Labs\\PuppetInstaller"
+            reg_query_command = %Q(reg query "#{reg_key}" /v "RememberedPuppetAgentStartupMode" | findstr #{msi_opts['PUPPET_AGENT_STARTUP_MODE']})
```
Comment (Iristyle) [inline lib/beaker/dsl/install_utils/windows_utils.rb]: The value specified to the MSI is written exactly as passed, so the literal string comparison here should be fine.

👍

**Comment by Iristyle** [comment]
I'm 👍 to everything here.

**Comment by tvpartytonight** [comment]
Thanks for the PR @MosesMendoza , the merge into master will be released in the next beaker release, which will be 3.0.0.

**Pull request merged**